Bug: Type inference fails when assigning namespaced array to local variable from parent-path import

## Description

When a module uses a parent-path import (`../library`) and attempts to assign a namespaced array to a local variable, type inference fails and the variable's type becomes empty/unknown.

## Reproducer

**Directory structure:**
```
project/
├── main.ez
└── src/
    ├── cli/
    │   └── cli.ez
    └── library/
        └── constants.ez
```

**src/library/constants.ez:**
```ez
module library

const validKeys [string, 2] = {
    "armadillo",
    "a8"
}
```

**src/cli/cli.ez:**
```ez
module cli

import @std
import lib"../library"

using std

do test_foreach() {
    temp keys = lib.validKeys
    for_each vk in keys {
        println(vk)
    }
}
```

**main.ez:**
```ez
import @std
import "./src/cli"

using std

do main() {
    cli.test_foreach()
}
```

## Error Output

```
error[E3017]: for_each requires array or string, got 
  --> src/cli/cli.ez:16:20
   |
16 |     for_each vk in keys {
   |                    ^ value is not iterable
```

Note: The error message shows "got " with nothing after it, indicating the type is empty/unknown.

## Expected Behavior

`temp keys = lib.validKeys` should correctly infer the type as `[string, 2]` (or `[string]`), and the subsequent for_each should work.

## Notes

- Related to #1110
- This works correctly when the imported module is in the same directory (`./mylib`)
- The bug appears to be specific to parent-path imports in nested modules
- Reproducer added to `integration-tests/fail/multi-file/foreach-parent-import/`

--- a/cmd/ez/main.go
+++ b/cmd/ez/main.go
@@ -922,6 +922,11 @@ func runFile(filename string) {
 				fileTc.RegisterModuleFunction(modName, funcName, sig)
 			}
 		}
+		for modName, vars := range moduleVariables {
+			for varName, varType := range vars {
+				fileTc.RegisterModuleVariable(modName, varName, varType)
+			}
+		}
 
 		// For multi-file modules, register module-internal types/functions/variables (#722)
 		if pf.numFiles > 1 {

